dojo/tools/dependency_track/parser.py

```diff
@@ -242,16 +345,15 @@ def get_description_for_scan_types(self, scan_type):
         return "The Finding Packaging Format (FPF) from OWASP Dependency Track can be imported in JSON format. See here for more info on this JSON format."
 
     def get_findings(self, file, test):
-
         # Exit if file is not provided
         if file is None:
             return list()
 
         # Load the contents of the JSON file into a dictionary
         data = file.read()
         try:
-            findings_export_dict = json.loads(str(data, 'utf-8'))
-        except:
+            findings_export_dict = json.loads(str(data, "utf-8"))
+        except BaseException:
```

Comment:
Is this an accurate exception?

Reply:
Hi @Maffooch, flake8 applies this change to comply with this [rule](https://www.flake8rules.com/rules/E722.html) and according to the [BaseException](https://docs.python .org/ 3/library/exceptions.html#BaseException) documentation  is a base class for any exception.

Reply:
Please change the use of `BaseException` to the standard catch all `Exception`

Reason for this is that the `BaseException` class will also catch system exits and keyboard interrupts that may be not be desired/expected. The python docs you linked also mention that the `BaseException` class should not be directly extended (and therefore not directly used either) but that the `Exception` be used instead.

> The base class for all built-in exceptions. It is not meant to be directly inherited by user-defined classes (for that, use [Exception](https://docs.python.org/3/library/exceptions.html#Exception)).